src/game/Unit.cpp

```diff
@@ -2099,69 +2099,91 @@ MeleeHitOutcome Unit::RollMeleeOutcomeAgainst(const Unit* pVictim, WeaponAttackT
 
     // bonus from skills is 0.04%
     int32 skillBonus  = 4 * (attackerWeaponSkill - victimMaxSkillValueForLevel);
-    int32 sum = 0;
-    int32 roll = urand(0, 10000);
-    int32 tmp = miss_chance;
+    int32 roll = urand(0, 100) * 100;
 
     DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: skill bonus of %d for attacker", skillBonus);
     DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: rolled %d, miss %d, dodge %d, parry %d, block %d, crit %d",
                      roll, miss_chance, dodge_chance, parry_chance, block_chance, crit_chance);
 
-    if (tmp > 0 && roll < (sum += tmp))
+    bool from_behind = !pVictim->HasInArc(M_PI_F, this);
+
+    if (from_behind)
+        DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: attack came from behind.");
+
+    if (roll < miss_chance)
     {
-        DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: MISS");
+        DEBUG_FILTER_LOG(LOG_FILTER_COMBAT, "RollMeleeOutcomeAgainst: MISS %d", miss_chance);
         return MELEE_HIT_MISS;
     }
 
```

Comment:
I was actually in doubt about that, since I have seen both practises throughout the code. I will stick to not having spaces between any if and else (and else if) from now on. (will of course also change any prs pending)